### ISSUE TITLE
No Ethernet detected on Raspberry Pi after updates.

### ISSUE THREAD

**Issue by foldfree** [void-linux/void-packages#6293]
Linux void-live 4.14.90_1 aarch64 musl

Running on the live version of 20181111, it works fine, however after update and reboot, no more ethernet. This is what I got from dmesg.

[    2.775190] lan78xx 1-1.1.1:1.0 (unnamed net_device) (uninitialized): can't register MDIO bus
[    2.775392] lan78xx 1-1.1.1:1.0 (unnamed net_device) (uninitialized): MDIO INIT FAILED.....
[    2.775442] lan78xx 1-1.1.1:1.0 (unnamed net_device) (uninitialized): Bind routine FAILED

ifconfig show no ethernet interface, so the kernel failed to load it.

I encounter one another person with the same issue on IRC.
any help kindly appreciated.

**Comment by foldfree** [comment]
https://github.com/raspberrypi/firmware/issues/957

I think it's this issue, any chance this gets patched locally?

**Comment by newbluemoon** [comment]
Just wanted to state that my RPi 3 is on armv7l-musl, up-to-date, and there everything works fine. Cannot test aarch64* (yet).

**Comment by Arcturus615** [comment]
Figured I would add to this. I'm also having the same issue following updating on a fresh install. The image used was the `void-rpi3-20181111` image, and dmesg is reporting the same error. It seems to have been reported almost nowhere, though I was able to find someone on [Alpine](https://bugs.alpinelinux.org/issues/9732)  that was having same issue in december. Does anyone know either how to solve this, or whats going on here?
```
[    2.779263] lan78xx 1-1.1.1:1.0 (unnamed net_device) (uninitialized): can't register MDIO bus
[    2.779569] lan78xx 1-1.1.1:1.0 (unnamed net_device) (uninitialized): MDIO INIT FAILED.....
[    2.779640] lan78xx 1-1.1.1:1.0 (unnamed net_device) (uninitialized): Bind routine FAILED
```

**Comment by Hoshpak** [comment]
Please install the new version of the rpi-firmware package and check if it improves the situation.

**Comment by foldfree** [comment]
I've tried raspbian, the official raspberry foundation image, and encountered the same issue.

**Comment by kervel** [comment]
Update: I built the 4.19 kernel from raspberry GitHub and the issue is gone here.

**Comment by Hoshpak** [comment]
so it's not about the firmware files. Maybe one of you who has an actual device to be able to possibly provide further pieces of information could report the issue upstream? If it's reproducable with raspbian, it should be much easier for upstream developers to debug. I don't know when we will be switching to the 4.19 kernel and if we don't know where the issue lies, there's no way to patch it for us.

**Comment by lpvm** [comment]
My aarch64 installation on a RPi3B+ is also affected, using 4.14.93_1.   Hope it gets fixed soon.

**Comment by Hoshpak** [comment]
Can anyone confirm that this is resolved with the update to 4.19?

**Comment by stewpend0us** [comment]
Worked for me
